lib/member_page.rb

```diff
@@ -15,7 +15,7 @@ class MemberPage < Scraped::HTML
   end
 
   field :name do
-    name_parts.first.to_s.gsub(Regexp.union(titles << '.'), '').tidy
+    name_parts.first.split.reject { |a| titles.include? a }.map(&:tidy).join(' ')
```

Comment:
the logic here is quite odd.

1. Why are you calling `tidy` _after_ the `reject`? If the parts aren't already suitable tidied, then they won't be correctly filtered out. If they _are_, then why re-tidy them again? What's that protecting against?

2. `name_parts.first.split` is a little hard to follow, possibly as `name_parts` doesn't really convey what the data is you're dealing with (and worse makes it sound like it's already the parts of the name, in which case why are we splitting them up again). I suspect it would be better to factor this out as another method, and make sure both have suitably descriptive names.

3. Iterating over a list to remove, one by one, the members of another list, is quite long-winded, confusing, and inefficient. `list1 - list2` is much simpler and clearer.